backend/fms_core/filters.py

```diff
@@ -0,0 +1,58 @@
+from django.db.models import Q
+
+from .models import Container, Sample, Individual
+
+import django_filters
+
+from .viewsets._constants import (
+    _container_filterset_fields,
+    _sample_filterset_fields,
+    _individual_filterset_fields,
+)
+
+class ContainerBatchFilter(django_filters.FilterSet):
+    barcode = django_filters.CharFilter(field_name="barcode", method="batch_barcode_filter")
+    name = django_filters.CharFilter(field_name="name", method="batch_name_filter")
+
+    class Meta:
+        model = Container
+        fields = _container_filterset_fields
+
+    def batch_barcode_filter(self, queryset, name, value):
+        query = Q()
+        for barcode in value.split(" "):
+            query |= Q(barcode__contains=barcode)
```

Comment:
This is a promising avenue ... To keep thing simple maybe you can load the "batch" filterset just when you have a batch request ... that way you may just use __in instead of __contains and ignore the other type of comparison in the batch filterset. The rest of the logic would remain the same for other fields. This would be a patch but this might prevent you from having to recode the whole filterset_field logic in your filterset.

Reply:
Mmm this may already be doing that for those 2 fields ... You'll still have to use a better comparison than contains for those batch fields.